### PR TITLE
HPCC-18518 Improve ws_FsService exception error message for missing Drop Zone.

### PR DESCRIPTION
**Pull request by AttilaVamos**
Extend exception error message.

Signed-off-by: Attila Vamos <attila.vamos@gmail.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Testing:

Tested manually with DFUPlus.

### CODE REVIEW

**Comment by AttilaVamos** [comment]
@richardkchapman please review and check whether the target is appropriate.

**Comment by richardkchapman** [comment]
@rpastrana Please review

**Review by rpastrana — CHANGES_REQUESTED** [review]
@AttilaVamos one suggestion.

**Comment by richardkchapman** [comment]
There's a typo in the commit message (Exted)

**Comment by AttilaVamos** [comment]
@rpastrana I can't see your suggestion.

**Comment by rpastrana** [comment]
@AttilaVamos I'm not sure what happened to my original message, here it is again:
At the point where the exception in thrown, we don't know if the error is due to an address or a path mismatch. Also, it might be helpful to point users to configuration.
"No dropzone configured for <IP> and <Path>. Check your system dropzone configuration"

**Comment by AttilaVamos** [comment]
@rpastrana Actually the CLocalEnvironmnet doesn't provide more information about why the address and the path not match. Actually it provides a Boolean value only. Perhaps another JIRA needed to implement that if it is really necessary.
Error messages updated.

File: esp/services/ws_fs/ws_fsService.cpp [hunk 1]
```diff
@@ -2237,7 +2237,7 @@ void CFileSprayEx::getDropZoneInfoByIP(double clientVersion, const char* ip, con
         destFile.set(destFileIn);
         Owned<IConstDropZoneInfo> dropZone = constEnv->getDropZoneByAddressPath(ip, destFile.str());
         if (!dropZone)
-            throw MakeStringException(ECLWATCH_DROP_ZONE_NOT_FOUND, "Dropzone not found for network address %s.", ip);
+            throw MakeStringException(ECLWATCH_DROP_ZONE_NOT_FOUND, "Drop zone to target '%s' not found for network address '%s'. Check your system drop zone configuration.", destFile.str(), ip);
```
Comment (rpastrana) [inline esp/services/ws_fs/ws_fsService.cpp]: @AttilaVamos
"No dropzone configured for 'IP' and 'PATH'. Check your system dropzone configuration" seems more direct and to the point. @JamesDeFabia  what do you think?
Also, is it "Dropzone" or "drop zone" ?

**Comment by AttilaVamos** [comment]
@richardkchapman 'useDropZoneRestriction' flag handling added.

**Comment by AttilaVamos** [comment]
ESP server crashed. Something is missing.

**Comment by AttilaVamos** [comment]
@richardkchapman ESP crash fixed, please review.

**Pull request merged**